### PR TITLE
FOUNDATIONS: Add Artifact

### PR DESCRIPTION
**Pull request by mimendel**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -59,7 +59,7 @@ BenchmarkDotNet.Artifacts/
 # .NET Core
 project.lock.json
 project.fragment.lock.json
-artifacts/
+#artifacts/
```
Comment (AndrewMcClelland) [inline .gitignore:62]: We can maybe find a better way to do this (ignore all `artifacts/` except for the DMX ones? Or vice versa

File: DMX.Core.Api.Tests.Unit/Services/Foundations/Artifacts/ArtifactServiceTests.Exceptions.Add.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace DMX.Core.Api.Tests.Unit.Services.Foundations.Artifacts
+{
+    public partial class ArtifactServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnAddIfDependencyErrorOccursAndLogItAsync()
```
Comment (AndrewMcClelland) [inline DMX.Core.Api.Tests.Unit/Services/Foundations/Artifacts/ArtifactServiceTests.Exceptions.Add.cs:19]: Note for @mimendel-public and I - see if there are any other exceptions that the SDK throws

**Review by hassanhabib — DISMISSED** [review]

File: DMX.Core.Api.Tests.Unit/Services/Foundations/Artifacts/ArtifactServiceTests.Exceptions.Add.cs (hunk 2 above)
Comment (hassanhabib) [reply DMX.Core.Api.Tests.Unit/Services/Foundations/Artifacts/ArtifactServiceTests.Exceptions.Add.cs]: @AndrewMcClelland let's pair on this today if you have time.

File: .gitignore (hunk 1 above)
Comment (AndrewMcClelland) [reply .gitignore:62]: This should be ok for now

**Review by AndrewMcClelland — APPROVED** [review]

**Review by hassanhabib — APPROVED** [review]

**Pull request merged**